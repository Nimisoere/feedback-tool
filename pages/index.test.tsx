import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import mockRouter from 'next-router-mock';
import Index from "./index";

jest.mock('next/router', () => require('next-router-mock'));

describe("Feedback form page", () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl("/");
  });

  it("should match snapshot", async () => {
    const { getByText, container } = render(
      <Index />
    );
    expect(container).toMatchSnapshot();
  });
});

