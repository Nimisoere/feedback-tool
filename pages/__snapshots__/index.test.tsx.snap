// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback form page should match snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col items-stretch"
  >
    <header
      class="w-full sticky top-0 z-10 shadow bg-white sm:h-20 py-4 sm:py-0 h-auto flex items-center transition-all duration-500"
    >
      <div
        class="container mx-auto flex flex-wrap justify-between"
      >
        <div>
          <a
            class="font-bold text-2xl"
            href="/"
          >
            FEEDr.
          </a>
        </div>
        <nav
          class="gap-4 sm:w-auto w-full flex justify-between"
        >
          <a
            class="btn-plain"
            href="/"
          >
            Give feedback
          </a>
          <a
            class="btn-plain"
            href="/view-feedback"
          >
            View feedback
          </a>
        </nav>
      </div>
    </header>
    <main
      class="flex-grow"
    >
      <div
        class="py-10"
      >
        <div
          class="container max-w-3xl mx-auto"
        >
          <div
            class="mb-8"
          >
            <h1
              class="font-medium text-2xl"
            >
              Feedback Form
            </h1>
          </div>
          <div>
            <form>
              <div
                class="flex flex-row gap-8 flex-wrap sm:flex-nowrap"
              >
                <div
                  class="w-full sm:w-1/3"
                >
                  <div
                    class="flex flex-col"
                  >
                    <div
                      class="flex flex-col mb-8"
                    >
                      <label
                        class="font-medium text-sm"
                      >
                        Name
                      </label>
                      <input
                        class="border-slate-300 focus:outline-cyan-400 bg-slate-50 border-2 rounded-md py-2 px-4 "
                        id="name"
                        name="name"
                        placeholder="Enter your name"
                        type="text"
                      />
                    </div>
                    <div
                      class="flex flex-col mb-8"
                    >
                      <label
                        class="font-medium text-sm"
                      >
                        Email
                      </label>
                      <input
                        class="border-slate-300 focus:outline-cyan-400 bg-slate-50 border-2 rounded-md py-2 px-4 "
                        id="email"
                        name="email"
                        placeholder="Enter your email"
                        type="email"
                      />
                    </div>
                    <div
                      class="flex flex-col mb-8"
                    >
                      <label
                        class="font-medium text-sm"
                      >
                        Rating
                      </label>
                      <div
                        class="flex flex-wrap sm:flex-col"
                      >
                        <div
                          class="text-slate-400 w-full"
                        >
                          <input
                            class="hidden peer"
                            id="range-1-rating"
                            name="rating"
                            placeholder="Enter your rating"
                            type="radio"
                            value="1"
                          />
                          <label
                            class="cursor-pointer text-sm peer peer-checked:text-yellow-400"
                            for="range-1-rating"
                          >
                            <span
                              class="text-2xl"
                            >
                              ⭑
                            </span>
                             (
                            1
                             
                            star
                            )
                          </label>
                        </div>
                        <div
                          class="text-slate-400 w-full"
                        >
                          <input
                            class="hidden peer"
                            id="range-2-rating"
                            name="rating"
                            placeholder="Enter your rating"
                            type="radio"
                            value="2"
                          />
                          <label
                            class="cursor-pointer text-sm peer peer-checked:text-yellow-400"
                            for="range-2-rating"
                          >
                            <span
                              class="text-2xl"
                            >
                              ⭑
                            </span>
                             (
                            2
                             
                            stars
                            )
                          </label>
                        </div>
                        <div
                          class="text-slate-400 w-full"
                        >
                          <input
                            class="hidden peer"
                            id="range-3-rating"
                            name="rating"
                            placeholder="Enter your rating"
                            type="radio"
                            value="3"
                          />
                          <label
                            class="cursor-pointer text-sm peer peer-checked:text-yellow-400"
                            for="range-3-rating"
                          >
                            <span
                              class="text-2xl"
                            >
                              ⭑
                            </span>
                             (
                            3
                             
                            stars
                            )
                          </label>
                        </div>
                        <div
                          class="text-slate-400 w-full"
                        >
                          <input
                            class="hidden peer"
                            id="range-4-rating"
                            name="rating"
                            placeholder="Enter your rating"
                            type="radio"
                            value="4"
                          />
                          <label
                            class="cursor-pointer text-sm peer peer-checked:text-yellow-400"
                            for="range-4-rating"
                          >
                            <span
                              class="text-2xl"
                            >
                              ⭑
                            </span>
                             (
                            4
                             
                            stars
                            )
                          </label>
                        </div>
                        <div
                          class="text-slate-400 w-full"
                        >
                          <input
                            class="hidden peer"
                            id="range-5-rating"
                            name="rating"
                            placeholder="Enter your rating"
                            type="radio"
                            value="5"
                          />
                          <label
                            class="cursor-pointer text-sm peer peer-checked:text-yellow-400"
                            for="range-5-rating"
                          >
                            <span
                              class="text-2xl"
                            >
                              ⭑
                            </span>
                             (
                            5
                             
                            stars
                            )
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="w-full sm:w-2/3"
                >
                  <div
                    class="flex flex-col mb-8"
                  >
                    <label
                      class="font-medium text-sm"
                    >
                      Comment
                    </label>
                    <textarea
                      class="border-slate-300 focus:outline-cyan-400 bg-slate-50 border-2 rounded-md py-2 px-4 "
                      id="comment"
                      name="comment"
                      placeholder="Enter your comment"
                      rows="10"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex justify-end"
              >
                <button
                  class="btn-primary w-full sm:w-auto"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
